### ISSUE TITLE
Postal Code Ranges and US 5+4 Zip codes

### ISSUE THREAD

**Issue by asconsulting** [isotope/core#2236]
At issue is that US postal codes can be in the format of XXXXX-XXXX. The Tax Rates and Shipping Models parse the Zip codes via Frontend::parsePostalCodes(); Because these fields accept ranges using a dash delimiter odd behavior can occur when a 5+4 zip code is entered. 

I have created an extension that works around this by changing the range delimiter from a dash (-) to a tilde (~). Because changing this behavior would break many existing Isotope installations using Postal Code lookups, I have added a config variable to tl_iso_config that defaults to using the legacy logic.

You can see my workaround for this issue here:
https://github.com/asconsulting/isotope_us_postal_codes

**Comment by aschempp** [comment]
Nice! Do I understand correctly that ranges to not work with these kind of postal codes?

**Comment by asconsulting** [comment]
Yes, there are some areas where it seems we may have a false match. Or more commonly no match because they get broken up and treated as a range.

For example 01002-9231 Is a valid zip code, but by basic range math it could cover zips 01002 through 09231. But more commonly we have a 5+4 zip setup as a tax rate, and then the end user address doesn't match because it is not checking the whole string.

Also, I don't think you could get a range to work with these zip code using PHP range function as is. Essentially the +4 acts as a sub code under the main zip. So I suppose an ideal solution would ignore the +4 for calculating if something is in range. So 01002 should accept anything from 01002-0000 through 01002-9999 as well as just plain 01002.  So then a range of 01002~01006 would accept 01002, 01003, 01004, 01005, 01006 with any +4 added to it. 

We would just trim them off and be done with it, but UPS and FedEx address validation returns these validated addresses with the 5+4 zips.

**Comment by aschempp** [comment]
> Essentially the +4 acts as a sub code under the main zip.

So if you do a ZIP range, would you expect to narrow it down to the main ZIP only (range of main ZIP) or rather everything in one main ZIP? Or even just a range of sub code in a main zip?

**Comment by asconsulting** [comment]
I don't think that US taxes are ever specific enough to need the +4 digits. So a range of zip codes ideally would be inclusive (and ignore) the +4. So I would say it would just have to match the main zip.
